benchmarks/Csharp/Benchmarks/NBody.cs

```diff
@@ -0,0 +1,195 @@
+﻿namespace Harness.Benchmarks;
+
+public sealed class NBody : IBenchmark
```

Comment:
Is the `I` here required by naming convention?
Benchmark shouldn't be an interface but a class. 
There's a concrete method in the `Benchmark` class.

Reply:
I is naming convetion for an Interface, yes. But I can make Benchmark an abstract class (ideally generic in return type to avoid the object casting) and then it will be `Benchmark`.

Reply:
Hm, I'd be worried about the implications from a compilation perspective.

This seems to be a change from other implementations, and it's not clear to me that the departure is "necessary" in a sense.

AreWeFastYet only "works" because we are trying to be very consistent, and balance idiomatic constraints with comparability/fairness.
So, ideally, we give each language's compiler/runtime exactly the same challenges, minimizing any deviation that's not forced on us by language differences.

Reply:
From a compilers perspective, I think this is a virtual call (callvirt IL instruction) one way or the other. But I see that you are favoring consistency stringly and will change this to be a 1:1 match with Java